parse-server-ascension/cloud/student.js

```diff
@@ -187,20 +187,41 @@ Parse.Cloud.define("AssignHouse", async(request) => {
     const query = new Parse.Query(House); 
     const argument = request.params;
     const res = await query.find();
-    
-    //Get lowest!
-    var randNum = Math.floor(Math.random() * res.length);
-    var pickedHouse = res[randNum];
 
+    //Note: All houses should begin with the same population count; base case (e.g [0,0,0,0])
+    const res2 = JSON.parse(JSON.stringify(res))
+    let max = 0
+    let counter = 0
+    for(let i = 0; i < res2.length; i++){
```

Comment:
You can use prebuilt functions in js

**Example**
var nums = [1, 2, 3]
Math.min.apply(Math, nums); //returns 1
Math.max.apply(Math, max); //returns 3